dialogue-annotations/src/main/java/com/palantir/dialogue/annotations/Request.java

```diff
@@ -53,6 +55,8 @@
      */
     Class<? extends DeserializerFactory> accept() default Json.class;
 
+    Class<? extends ErrorDecoder> errorDecoder() default ConjureErrorDecoder.class;
```

Comment:
@carterkozak I think this is what you described?

Reply:
so flows would be:
1. Say you hate exceptions, you'd impl ErrorDecoder and always return false from isError, and then all the handling would happen in DeserializerFactory.
2. Say you don't like conjure exceptions, you'd just impl your own ErrorDecoder.
3. Most people don't care about errors, so they don't need to do anything special.
4. There's some fun extra handling in ConjureBodySerDe around 204s, which might be useful to port over here?